PoseEstimation/poseEstimation_main.py

```diff
@@ -0,0 +1,182 @@
+import os, requests
+import sys
+import numpy as np
+import pandas as pd
+import h5py
+import typing
+from pathlib import Path
+import matplotlib.pyplot as plt
+from tifffile import imwrite, TiffWriter, TiffFile
+import yaml
+import cv2
+import copy
+from datetime import datetime
+import glob
+
+sys.path.append("..")
+import utilities as utils
+import definitions as defs
+import poseEstimation_parameters as poseEst_params
+import poseEstimation_definitions as poseEst_defs
+
+import deeplabcut
+print("Imported DLC!", flush=True)
+
+# Import for PyInstaller
+from multiprocessing import freeze_support
+freeze_support()
+
+def main(poseEstimation_params: poseEst_params.PoseEstimationParameters):
+
+    """
+    DeepLabCut algorithm
+    """
+    # --------------- Read danse parameters ---------------
+    Operations: str    = poseEstimation_params.params[defs.DoricFile.Attribute.Group.OPERATIONS]
+    filePath: str      = poseEstimation_params.paths[defs.Parameters.Path.FILEPATH]
+    datapath: str      = poseEstimation_params.paths[defs.Parameters.Path.H5PATH]
+    projectFolder: str = poseEstimation_params.params[poseEst_defs.Parameters.danse.PROJECT_FOLDER]
+    bodyPartNames      = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_NAMES].split(', ')
+    bodyPartColors     = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_COLORS].split(', ')
+    extractedFrames    = poseEstimation_params.params[poseEst_defs.Parameters.danse.EXTRACTED_FRAMES]
+    trainingCoordinates = {}
+    for bodyPart in bodyPartNames:
+        label = bodyPart + poseEst_defs.Parameters.danse.COORDINATES
+        trainingCoordinates[label] = poseEstimation_params.params[label]
+
+    # --------------- Create Project folder and config file ---------------
+    task: str        = "DLC" 
+    experimenter:str = "doric"
+
+    file_ = h5py.File(filePath, 'a')
+    attributes = utils.load_attributes(file_, datapath)
+    key = [key for key in attributes if poseEst_defs.Parameters.danse.VIDEO_FILEPATH in key]   
+    videoPath: str = attributes[key[0]]
+    # file_.close
+
+    path_config_file: str = deeplabcut.create_new_project(task, experimenter, [videoPath], projectFolder, copy_videos = False)
+    updateConfigFile(path_config_file, bodyPartNames)
+
+    # --------------- Create hdf file for labeled data ---------------
+    createlabeledDataHDF(path_config_file, extractedFrames, bodyPartNames, experimenter, poseEstimation_params, videoPath)
+
+    # --------------- Create a training dataset ---------------
```

Comment:
I don't feel like comments that referring to only 1 line of code are useful, especially if they just repeat the name of the function. I'd remove them

Reply:
Here we have comments to 1 line of code, and below there're chunks of code without any explanation..